backend/fms_core/services/library.py

```diff
@@ -48,3 +51,96 @@ def create_library(library_type, index, platform, strandedness, library_size=Non
         errors.append(';'.join(e.messages))
 
     return library, errors, warnings
+
+
+def convert_library(process, platform, sample_source, concentration, library_size, container_destination,
+                    coordinates_destination, volume_used, volume_destination, execution_date, comment):
+    library_destination = None
+    errors = []
+    warnings = []
+
+    if not process:
+        errors.append(f"Process is required.")
+    if not platform:
+        errors.append(f"Platform is required.")
+    if not library_size:
+        errors.append(f"Library size is required.")
+    if not sample_source:
+        errors.append(f"Source sample is required.")
+    if not container_destination:
+        errors.append(f"Destination container is required.")
+    if volume_used is None:
+        errors.append(f"Volume used is required.")
+    else:
+        if volume_used <= 0:
+            errors.append(f"Volume used ({volume_used}) is invalid.")
+        if sample_source and volume_used > sample_source.volume:
+            errors.append(
+                f"Volume used ({volume_used}) exceeds the current volume of the library ({sample_source.volume}).")
+
+    if not isinstance(execution_date, date):
+        errors.append(f"Execution date is not valid.")
+
+    # Retrieve library object linked to the source sample
+    library_source_obj = sample_source.derived_sample_not_pool.library
```

Comment:
We are about to deal with pool and this will happen soon. Let's assume a library conversion can be applied to a pool an create a library for each derived_sample.

Reply:
This is still not applied to a pool. I know this is not a case now without pools but it will soon enough and this is much simpler to fix while you have the code fresh in mind instead of in 2 months where we may forget it.